Chocolate_Vending_Machine.java

```diff
@@ -0,0 +1,240 @@
+			while (diffMoney > 0) {
+				if (diffMoney >= Coin.ten.getDenomination() && cash.get(Coin.ten) >= coindeduct[0]) {
+					diffMoney -= Coin.ten.getDenomination();
+					coindeduct[0]++;
+					changes.add(Coin.ten);
+				} else if (diffMoney >= Coin.five.getDenomination() && cash.get(Coin.five) >= coindeduct[1]) {
+					diffMoney -= Coin.five.getDenomination();
+					coindeduct[1]++;
+					changes.add(Coin.five);
+				} else if (diffMoney >= Coin.two.getDenomination() && cash.get(Coin.two) >= coindeduct[2]) {
+					diffMoney -= Coin.two.getDenomination();
+					coindeduct[2]++;
+					changes.add(Coin.two);
+				} else if (diffMoney >= Coin.one.getDenomination() && cash.get(Coin.two) >= coindeduct[3]) {
+					diffMoney -= Coin.one.getDenomination();
+					coindeduct[3]++;
+					changes.add(Coin.one);
+				} else {
+					throw new NoSufficientChangeException("Machine does not have sufficient change");
+				}
+			}
+		}
+		return changes;
+	}
+
+	private void updateCashInventory(ArrayList<Coin> changes) {
+		for (Coin c : changes) {
+			removeCoin(c, 1);
+		}
+	}
+
+	private void updateItemInventory(int s_no, int qty) {
+		Item items_in[] = Item.values();
+		removeItem(items_in[s_no - 1], qty);
+
+	}
+
+	public void collectMoney_And_UpdateInventories(ArrayList<Coin> changes, int s_no, int qty) {
+		Item items_in[] = Item.values();
+		System.out.println("Collect your money!! ");
+		if (changes.size() > 0) {
+			for (Coin c : changes) {
+				System.out.print(c.getDenomination() + " ");
+			}
+			System.out.println();
+		} else {
+			System.out.println("No return change left");
+		}
+		System.out.println("Your order ");
+		System.out.println("Item                      Quantity");
+		System.out.println(items_in[s_no - 1].getName(items_in[s_no - 1]) + "                	" + qty);
+		System.out.println();
+		System.out.println("Please Visit Again");
+		updateCashInventory(changes);
+		updateItemInventory(s_no, qty);
+	}
+
+	public void printInventoryStatus() {
+		Item items_in[] = Item.values();
+		for (Item i : items_in) {
+			System.out.println(i.getName(i) + "     " + items.get(i));
+		}
+	}
+
+	public static void main(String[] args) {
+		try {
+			Scanner s = new Scanner(System.in);
+			Chocolate_Vending_Machine vm = new Chocolate_Vending_Machine();
+			vm.displayMenu();
+
+			int s_no = s.nextInt();
+			if (!(s_no >= 1 && s_no <= 4)) {
+				throw new InvalidItemNoEntered("Wrong serial no of the item entered");
+			}
+			int qty = s.nextInt();
+			if (!vm.hasSufficientItem(s_no, qty)) {
+				throw new OutOfStockException("Sorry!! We don't have enough chocolates of your demand");
+			}
+
+			ArrayList<Integer> money = new ArrayList<>();
+			System.out.println("Start entering money and enter -1 to halt");
+			int coin = 0;
+			do {
+				coin = s.nextInt();
+				if(!(coin!=-1||coin!=1||coin!=2||coin!=5||coin!=10) ){
+					throw new NotAcceptableDenominationEntered("This last inserted coin's value is not acceptable");
+				}
+				money.add(coin);

```

Comment:
it looks like you're adding -1 in this case.  Is that ok?

Reply:
Also, is this how a vending machine works?  I could enter money, try to make a selection (it fails) enter more money, make a selection

Reply:
But how can I make it  to wait for user input always?